pylops/LinearOperator.py

```diff
@@ -253,12 +253,32 @@ def dot(self, x):
         elif np.isscalar(x):
             return _ScaledLinearOperator(self, x)
         else:
-            if x.ndim == 1:  # or x.ndim == 2 and x.shape[1] == 1:
-                return self.matvec(x)
+            dims = getattr(self, "dims", None)
+            dimsd = getattr(self, "dimsd", None)
+
+            reshape_out = False
```

Comment:
Can you please add a comment here, explaining what this piece of code does... that it will reshape the output back into a ndarray of size dimsd when the input had shape dims